fluent-jdbc/src/main/java/org/codejargon/fluentjdbc/internal/mappers/ObjectMapper.java

```diff
@@ -94,7 +94,7 @@ private Object optionalOf(Field field, Object value) {
             return Optional.empty();
         } else {
             Class<?> typeOfField = typeOfField(field);
-            if (!typeOfField.isAssignableFrom(value.getClass())) {
+            if (!value.getClass().isAssignableFrom(typeOfField)) {
```

Comment:
This is a bug. It will fail if value from JDBC is java.sql.Date and the target field is Optional<java.util.Date> 
Fixed this in the branch and extended the test so it checks if such assignments are ok.